src/ZoneServer/Skills/Handlers/Swordsman/Hoplite/SpearLunge.cs

```diff
@@ -0,0 +1,124 @@
+
+namespace Melia.Zone.Skills.Handlers.Swordsman.Hoplite
+{
+	/// <summary>
+	/// Handler for the Hoplite skill Spear Lunge.
+	/// </summary>
+	[SkillHandler(SkillId.Hoplite_SpearLunge)]
+	public class Hoplite_SpearLunge : IGroundSkillHandler
+	{
+		/// <summary>
+		/// Handles skill, damaging targets.
+		/// </summary>
+		/// <param name="skill"></param>
+		/// <param name="caster"></param>
+		/// <param name="originPos"></param>
+		/// <param name="farPos"></param>
+		public void Handle(Skill skill, ICombatEntity caster, Position originPos, Position farPos, ICombatEntity target)
+		{
+			if (!caster.TrySpendSp(skill))
+			{
+				caster.ServerMessage(Localization.Get("Not enough SP."));
+				return;
+			}
+
+			skill.IncreaseOverheat();
+			caster.TurnTowards(farPos);
+			caster.SetAttackState(true);
+
+			var splashParam = skill.GetSplashParameters(caster, originPos, farPos, length: 70, width: 20, angle: 0);
+			var splashArea = skill.GetSplashArea(SplashType.Square, splashParam);
+
+			Send.ZC_SKILL_READY(caster, skill, originPos, farPos);
+			Send.ZC_SKILL_MELEE_GROUND(caster, skill, farPos, null);
+
+			this.Attack(skill, caster, splashArea);
+		}
+
+		/// <summary>
+		/// Executes the actual attack after a delay.
+		/// </summary>
+		/// <param name="skill"></param>
+		/// <param name="caster"></param>
+		/// <param name="splashArea"></param>
+		private async void Attack(Skill skill, ICombatEntity caster, ISplashArea splashArea)
+		{
+			var hitDelay = TimeSpan.FromMilliseconds(100);
+			var damageDelay = TimeSpan.FromMilliseconds(50);
+			var delayBetweenHits = TimeSpan.FromMilliseconds(100);
+			var skillHitDelay = TimeSpan.Zero;
+
+			await Task.Delay(hitDelay);
+
+			var targets = caster.Map.GetAttackableEntitiesIn(caster, splashArea);
+			var hits = new List<SkillHitInfo>();
+
+			foreach (var target in targets.LimitBySDR(caster, skill))
+			{
+				var modifier = SkillModifier.Default;
+				modifier.DefensePenetrationRate = 0.15f;
+
+				var skillHitResult = SCR_SkillHit(caster, target, skill, modifier);
+
+				if (skillHitResult.Result == HitResultType.Crit && caster.Components.TryGet<SkillComponent>(out var skills)
+					&& skills.TryGet(SkillId.Hoplite_SharpSpear, out var sharpSpear))
+				{
+					skillHitResult.Damage += skillHitResult.Damage *= (0.1f + sharpSpear.Level * 0.02f);
+				}
+
+				target.TakeDamage(skillHitResult.Damage, caster);
+
+				var skillHit = new SkillHitInfo(caster, target, skill, skillHitResult, damageDelay, skillHitDelay);
+				skillHit.HitEffect = HitEffect.Impact;
+
+				hits.Add(skillHit);
+			}
+
+			Send.ZC_SKILL_HIT_INFO(caster, hits);
+
+			await Task.Delay(delayBetweenHits);
+			hits.Clear();
+			targets = caster.Map.GetAttackableEntitiesIn(caster, splashArea);
+
+			foreach (var target in targets.LimitBySDR(caster, skill))
+			{
+				var modifier = SkillModifier.Default;
+				modifier.DefensePenetrationRate = 0.15f;
+
+				var skillHitResult2 = SCR_SkillHit(caster, target, skill, modifier);
```

Comment:
There isn't really any reason to append a `2` to this variable name, as the two results are both in their own scope. Generally you would do this if they're in the same scope, which means this naming convention conveys something that isn't the case.

Reply:
True.